docs/philosophy.rst

```diff
@@ -0,0 +1,183 @@
+
+(Mixed-Integer) Linear Programming
+==================================
+
+Much of the mathematical optimization done today is done with
+`linear programming <https://en.wikipedia.org/wiki/Linear_programming>`_
+solvers and techniques.
+
+Users familiar with linear programming techniques will find many concepts in
+`dwave-optimization` familiar: objective functions; continuous, linear, and binary variables;
+constraints; and feasible regions are all concepts found in linear programming.
+
+However, unlike linear solvers, `dwave-optimization` allows
+`nonlinear <https://en.wikipedia.org/wiki/Nonlinear_programming>`_ relationships
+among variables, constraints, and the objective function.
+
+.. _optimization_philosophy_combinatorial_variables:
+
+Lists, Sets, and other combinatorial variables
+==============================================
+
+While many optimization problems can be expressed using only scalar variable
+types (continuous, integer, and binary), it is often convenient to use other
+combinatorial structures.
+
+In the following example, we will contrast a classic mixed integer linear programming
+formulation using scalar variables with one that uses a list variable.
+
+Example - Traveling Salesperson
+-------------------------------
+
+The goal of renowned
+`traveling salesperson problem <https://en.wikipedia.org/wiki/Travelling_salesman_problem>`_
+is, for a given a list of cities and distances between each pair of cities, to
+find the shortest possible route that visits each city exactly once and returns
+to the city of origin.
+
+If we wish to restrict ourselves to scalar variables,
+a common mixed-integer linear programming formulation for solving a TSP with
+:math:`N` cities is to use :math:`N^2` binary variables and :math:`2N`
+constraints.
+
+.. dropdown:: MILP formulation details
+
+    Let
+
+    .. math::
+        
+        x_{it} =
+            \begin{cases} 
+                1 \text{ if the salesperson is in city } i \text{ at time } t \\
+                0 \text{ otherwise} \\
+            \end{cases}
+
+    If :math:`d_{i,j}` gives the distance from city :math:`i` to city :math:`j` then
+    the objective is to minimize the total tour length
+
+    .. math::
+
+        \sum_{st} \sum_{ij} x_{is} x_{jt} d_{ij}
+
+    subject to constraints that the salesperson cannot be in two cities
+    at the same time, nor any city twice
+
+    .. math::
+
+        \sum_i x_{it} = 1 \text{  } \forall t \\
+        \sum_t x_{it} = 1 \text{  } \forall i
+
+However, the solution to this problem is defined by an ordered list of cities.
+For example, a three city traveling salesperson problem has
+`(0, 1, 2)`, `(0, 2, 1)`, `(1, 0, 2)`, `(1, 2, 0)`, `(2, 0, 1)`, and `(2, 1, 0)`
+as possible solutions.
+Therefore, rather than thinking about scalars, we can define a *list* variable as
+any permutation of the integers from :math:`[0..N)`.
+Another advantage of this approach is that the problem is unconstrained!
```

Comment:
```suggestion
Note that the solution to this problem is defined by an ordered list of cities.
For example, a three-city TSP has these possible solutions: `(0, 1, 2)`,
`(0, 2, 1)`, `(1, 0, 2)`, `(1, 2, 0)`, `(2, 0, 1)`, and `(2, 1, 0)`.
Therefore, rather than working with scalars, you can define a *list* variable as
any permutation of the integers from :math:`[0..N)`, significantly reducing the
number of problem variables. Another advantage of this approach is that the
problem is unconstrained!
```

See the alternative below (links to `list()`)